### PR TITLE
Fix header parsing in JWT filter

### PR DESCRIPTION
**Pull request by yangminzhu**
Fix header parsing in JWT filter

### CODE REVIEW

File: src/envoy/http/jwt_auth/jwt.h [hunk 1]
```diff
@@ -231,8 +231,6 @@ class Jwt : public WithStatus {
   int64_t Exp();
 
  private:
-  const EVP_MD* md_;
```
Comment (silentdai) [inline src/envoy/http/jwt_auth/jwt.h]: I am surprised a pointer can be removed without touching destructor.
Could you elaborate who owns the object `md_` used to reference?

Comment (yangminzhu) [reply src/envoy/http/jwt_auth/jwt.h]: Does it matter who owns it? I thought it's a const pointer pointing to a static variable.

**Comment by yangminzhu** [comment]
@fpesce @duderino 
The test `proxy-presubmit.sh` seems getting stuck forever, I have tried to retest, will try again to see if it works.

File: src/envoy/http/jwt_auth/jwt.h (hunk 1 above)
Comment (yangminzhu) [reply src/envoy/http/jwt_auth/jwt.h]: Talked offline and we think this is not a concern, resolved.

Comment (silentdai) [reply src/envoy/http/jwt_auth/jwt.h]: Thank you for the explanation! `md_` may not be introduced at the beginning. 
/lgtm

Comment (silentdai) [reply src/envoy/http/jwt_auth/jwt.h]: Sorry I was heading to a meeting so I didn't complete what I should have pointed out: The `static variable` owns so `Jwt` doesn't need to free it. That's what I wanted to know. Thanks again!

**Comment by fpesce** [comment]
Fixes istio/istio#15084

**Pull request merged**